azimuth/config.py

```
@@ -407,16 +411,15 @@ class AzimuthConfig(
     @root_validator()
     def dynamic_language_config_values(cls, values):
         defaults = config_defaults_per_language[values["language"]]
-        if values["behavioral_testing"]:
-            neutral_token = values["behavioral_testing"].neutral_token
+        if behavioral_testing := values.get("behavioral_testing"):
+            neutral_token = behavioral_testing.neutral_token
             neutral_token.prefix_list = neutral_token.prefix_list or defaults.prefix_list
             neutral_token.suffix_list = neutral_token.suffix_list or defaults.suffix_list
-        syntax = values["syntax"]
-        syntax.spacy_model = syntax.spacy_model or defaults.spacy_model
-        syntax.subj_tags = syntax.subj_tags or defaults.subj_tags
-        syntax.obj_tags = syntax.obj_tags or defaults.obj_tags
-        if values["similarity"]:
-            similarity = values["similarity"]
+        if syntax := values.get("syntax"):
```

Even though syntax is never optional at the moment?

---

Yes, for the reason I gave [here](https://github.com/ServiceNow/azimuth/pull/474#discussion_r1123826242).

---

Ah ok. if I rephrase, do you mean that if there is a problem with one syntax field (like the value being out of bound), the syntax options will be set to None or something like that, which will then break the old line 418, hiding the original error 
with the syntax field? If so, can we think of some comments to add so we don't forget? Or is basically your test testing that we have the appropriate error message?

---

Yes, "hiding the original error" with a `KeyError` on old line 418.
My test ensures that we get the expected `ValidationError`, which is the only type of exception pydantic is supposed to return. A `KeyError` makes the test fail.

---

Awesome, I get it now :)